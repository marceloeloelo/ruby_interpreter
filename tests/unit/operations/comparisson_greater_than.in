# int vs int

1 > 2
puts(1 > 2)
2 > 1
puts(2 > 1)
2 > 2
puts(2 > 2)

# double vs double

1.1 > 2.1
puts(1.1 > 2.1)
2.1 > 1.1
puts(2.1 > 1.1)
2.1 > 2.1
puts(2.1 > 2.1)

# int vs double

1 > 2.1
puts(1 > 2.1)
1.1 > 2
puts(1.1 > 2)
1 > 1.0
puts(1 > 1.0)

# string vs string

"foo" > "bar"
puts("foo" > "bar")
"bar" > "foo"
puts("bar" > "foo")
"foo" > "foo"
puts("foo" > "foo")
"" > "foo"
puts("" > "foo")
"" > ""
puts("" > "")

# string vs int

"foo" > 1
puts("foo" > 1)
2 > "foo"
puts(2 > "foo")
2 > "2"
puts(2 > "2")

# string vs double

2.1 > "foo"
puts(2.1 > "foo")
"foo" > 1.1
puts("foo" > 1.1)
"1.1" > 1.1
puts("1.1" > 1.1)

# bool vs bool

true > false
puts(true > false)
true > true
puts(true > true)
false > true
puts(false > true)
false > false
puts(false > false)

# bool vs int

true > 2
puts(true > 2)
true > 1
puts(true > 1)
true > 0
puts(true > 0)
2 > true
puts(2 > true)
1 > true
puts(1 > true)
0 > true
puts(0 > true)

false > 2
puts(false > 2)
false > 1
puts(false > 1)
false > 0
puts(false > 0)
2 > false
puts(2 > false)
1 > false
puts(1 > false)
0 > false
puts(0 > false)

# bool vs nil

true > nil
puts(true > nil)
nil > true
puts(nil > true)

false > nil
puts(false > nil)
nil > false
puts(nil > false)

# bool vs string

true > "foo"
puts(true > "foo")
true > ""
puts(true > "")
"foo" > true
puts("foo" > true)
"" > true
puts("" > true)

false > "foo"
puts(false > "foo")
false > ""
puts(false > "")
"foo" > false
puts("foo" > false)
"" > false
puts("" > false)

# bool vs double

true > 2.1
puts(true > 2.1)
true > 1.0
puts(true > 1.0)
true > 0.0
puts(true > 0.0)
2.1 > true
puts(2.1 > true)
1.0 > true
puts(1.0 > true)
0.0 > true
puts(0.0 > true)

false > 2.1
puts(false > 2.1)
false > 1.0
puts(false > 1.0)
false > 0.0
puts(false > 0.0)
2.1 > false
puts(2.1 > false)
1.0 > false
puts(1.0 > false)
0.0 > false
puts(0.0 > false)

# nil vs nil

nil > nil
puts(nil > nil)

# nil vs int

2 > nil
puts(2 > nil)
1 > nil
puts(1 > nil)
0 > nil
puts(0 > nil)

nil > 2
puts(nil > 2)
nil > 1
puts(nil > 1)
nil > 0
puts(nil > 0)

# nil vs double

2.1 > nil
puts(2.1 > nil)
1.0 > nil
puts(1.0 > nil)
0.0 > nil
puts(0.0 > nil)

nil > 2.1
puts(nil > 2.1)
nil > 1.0
puts(nil > 1.0)
nil > 0.0
puts(nil > 0.0)

 
