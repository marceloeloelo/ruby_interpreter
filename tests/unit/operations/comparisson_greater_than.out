false
true
false
false
true
false
false
false
false
true
false
false
false
false
TypeError: Fixnum can't be coerced into String
